blocks/components/editable/index.js

```diff
@@ -188,18 +207,26 @@ export default class Editable extends wp.element.Component {
 	}
 
 	onNodeChange( { parents } ) {
-		const formats = parents.reduce( ( result, node ) => {
+		let alignment = null;
+		const formats = {};
+
+		parents.forEach( ( node ) => {
 			const tag = node.nodeName.toLowerCase();
 
 			if ( formatMap.hasOwnProperty( tag ) ) {
-				result[ formatMap[ tag ] ] = true;
+				formats[ formatMap[ tag ] ] = true;
 			}
 
-			return result;
-		}, {} );
+			if ( tag === 'p' ) {
```

Comment:
>This is not a restriction, that's where TinyMCE adds the alignment style.

Ah. Thinking maybe this could be the `forced_root_block` setting:

https://www.tinymce.com/docs/configure/content-filtering/#forced_root_block

i.e. comparing to `this.editor.settings.forced_root_block` instead.